src/dao-proposals/dao-proposals.controller.ts

```diff
@@ -83,13 +86,26 @@ export class DaoProposalsController {
     }
 
     @Get(':id')
+    @ApiNoContentResponse()
     @ApiOkResponse({
         type: DaoProposal,
         description: 'Returns a single Proposal',
     })
     async getDaoProposalById(@Param('id') id: string): Promise<DaoProposal> {
         try {
-            return await this.getDaoProposalByIdService.execute(id);
+            const proposal = await this.getDaoProposalByIdService.execute(id);
+
+            if (!proposal) {
+                throw new HttpException(
+                    {
+                        status: HttpStatus.NO_CONTENT,
```

Comment:
question: shouldn't this be `HttpStatus.NOT_FOUND`? Like there is no proposal with that id we should clearly state that this is an issue because `NO_CONTENT` reflects to 204 which is inside http's success range.

Reply:
I guess it depends how you want to interpret it. For me 404 means that the route is not provided. 204 means that we do not have content for the id but were able to process the call. https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/204 as reference

Reply:
404 also states that the path is valid but the resource behind was not found. I will change it so you don't have to change the frontend.